src/shogun/clustering/KMeansBase.cpp

```diff
@@ -321,7 +320,9 @@ void KMeansBase::init()
 	SG_ADD(
 	    &use_kmeanspp, "kmeanspp", "Whether to use kmeans++",
 	    ParameterProperties::HYPER | ParameterProperties::SETTING);
-	SG_ADD(&mus, "mus", "Cluster centers", ParameterProperties::MODEL);
+	SG_ADD(
+	    &cluster_centers, "cluster_centers", "Cluster centers",
+	    ParameterProperties::MODEL);
 
-	watch_method("cluster_centers", &KMeansBase::get_cluster_centers);
```

Comment:
I would replace the SG_ADD above with the watch_method. And then you should have another SG_ADD for initial cluster centres.